controller/tests/utils/test_serial_comm.py

```diff
@@ -0,0 +1,738 @@
+        + TEST_INITIAL_MAGNET_FINDING_PARAMS["X"].to_bytes(1, byteorder="little", signed=True)
+        + TEST_INITIAL_MAGNET_FINDING_PARAMS["Y"].to_bytes(1, byteorder="little", signed=True)
+        + TEST_INITIAL_MAGNET_FINDING_PARAMS["Z"].to_bytes(1, byteorder="little", signed=True)
+        + TEST_INITIAL_MAGNET_FINDING_PARAMS["REMN"].to_bytes(2, byteorder="little", signed=True)
+        + bytes([is_stingray])
+        + convert_instrument_event_info_to_bytes(TEST_EVENT_INFO)
+    )
+
+    # TODO make sure the bytes are the correct length?
+
+    assert parse_metadata_bytes(metadata_bytes) == {
+        BOOT_FLAGS_UUID: 0b10101010,
+        MANTARRAY_SERIAL_NUMBER_UUID: TEST_SERIAL_NUMBER,
+        MANTARRAY_NICKNAME_UUID: "マンタレ1",
+        MAIN_FIRMWARE_VERSION_UUID: "0.1.2",
+        CHANNEL_FIRMWARE_VERSION_UUID: "255.255.255",
+        "status_codes_prior_to_reboot": convert_status_code_bytes_to_dict(bytes(test_status_codes)),
+        INITIAL_MAGNET_FINDING_PARAMS_UUID: TEST_INITIAL_MAGNET_FINDING_PARAMS,
+        "is_stingray": is_stingray,
+        **TEST_EVENT_INFO,
+    }
+
+
+@freeze_time("2021-04-07 13:14:07.234987")
+def test_get_serial_comm_timestamp__returns_microseconds_since_2021_01_01():
+    expected_usecs = (
+        datetime.datetime.now(tz=datetime.timezone.utc) - SERIAL_COMM_TIMESTAMP_EPOCH
+    ) // datetime.timedelta(microseconds=1)
+    actual = get_serial_comm_timestamp()
+    assert actual == expected_usecs
+
+
+@pytest.mark.parametrize(
+    "test_adc8,test_adc9,expected_impedance",
+    [(0, 0, 61.08021), (0, 2039, 21375.4744533), (0, 2049, -192709.2700799), (1113, 0, 9.9680762)],
+)
+def test_convert_adc_readings_to_impedance__returns_expected_values(test_adc8, test_adc9, expected_impedance):
+    actual_impedance = convert_adc_readings_to_impedance(test_adc8, test_adc9)
+    np.testing.assert_almost_equal(actual_impedance, expected_impedance)
+
+
+@pytest.mark.parametrize(
+    "test_impedance,expected_status",
+    [
+        (STIM_OPEN_CIRCUIT_THRESHOLD_OHMS + 1, StimulatorCircuitStatuses.OPEN),
+        (STIM_OPEN_CIRCUIT_THRESHOLD_OHMS, StimulatorCircuitStatuses.OPEN),
+        (STIM_OPEN_CIRCUIT_THRESHOLD_OHMS - 1, StimulatorCircuitStatuses.MEDIA),
+        (STIM_SHORT_CIRCUIT_THRESHOLD_OHMS + 1, StimulatorCircuitStatuses.MEDIA),
+        (STIM_SHORT_CIRCUIT_THRESHOLD_OHMS, StimulatorCircuitStatuses.SHORT),
+        (STIM_SHORT_CIRCUIT_THRESHOLD_OHMS - 1, StimulatorCircuitStatuses.SHORT),
+        (0, StimulatorCircuitStatuses.SHORT),
+        (-1, StimulatorCircuitStatuses.ERROR),
+    ],
+)
+def test_convert_adc_readings_to_circuit_status__returns_correct_values(
+    test_impedance, expected_status, mocker
+):
+    mocked_to_impedance = mocker.patch.object(
+        serial_comm, "convert_adc_readings_to_impedance", autospec=True, return_value=test_impedance
+    )
+    # mocking convert_adc_readings_to_impedance so these values don't actually matter
+    test_adc8 = randint(0, 0xFFFF)
```

Comment:
Just wondering, why is it necessary to use hex here?

Reply:
It's not, it just conveys intention more clearly